raft/src/test/java/org/apache/kafka/raft/RaftEventSimulationTest.java

```diff
@@ -966,6 +967,54 @@ public void verify() {
         }
     }
 
+    private static class SnapshotAtLogStart implements Invariant {
```

Comment:
Is there any noticeable change in simulation run time? It looks reasonable, but thought I'd check.

Reply:
Good catch. I made some improvements After the improvements the simulation time went from 1 minute in trunk to a 1:41 minutes for this PR.

Reply:
Okay. The simulations now take less than a minute in total on my machine which is what I see on trunk in my machine.